Failure was in random system test, PythonChannelTest
``` bash
======================================================================
ERROR: test_PythonLoggingChannel_update_instruments (PythonChannels.PythonLoggingTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/Testing/SystemTests/tests/framework/PythonChannels.py", line 87, in test_PythonLoggingChannel_update_instruments
    self._create_script_and_execute(LoggingClass="PythonLoggingChannel", UpdateInstrumentDefinitions=1)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/Testing/SystemTests/tests/framework/PythonChannels.py", line 99, in _create_script_and_execute
    result = subprocess.run([sys.executable, TEST_SCRIPT_NAME], timeout=60)  # This will raise subprocess.TimeoutExpired if deadlocke
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 505, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 1154, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 2047, in _communicate
    self.wait(timeout=self._remaining_time(endtime))
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 1209, in wait
    return self._wait(timeout=timeout)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.10/subprocess.py", line 1951, in _wait
    raise TimeoutExpired(self.args, timeout)
subprocess.TimeoutExpired: Command '['/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/bin/python3', 'pythonloggingtestscript.py']' timed out after 59.99998100101948 seconds

----------------------------------------------------------------------
```

--- a/Framework/DataHandling/src/Load.cpp
+++ b/Framework/DataHandling/src/Load.cpp
@@ -340,6 +340,9 @@ void Load::exec() {
 
   // Set the remaining properties of the loader
   setOutputProperties(m_loader);
+  // Set the loader and version from correct loader
+  setPropertyValue("LoaderName", m_loader->name());
+  setProperty("LoaderVersion", m_loader->version());
 }
 
 void Load::loadSingleFile() {

--- a/Framework/DataHandling/src/Load.cpp
+++ b/Framework/DataHandling/src/Load.cpp
@@ -340,9 +340,17 @@ void Load::exec() {
 
   // Set the remaining properties of the loader
   setOutputProperties(m_loader);
-  // Set the loader and version from correct loader
-  setPropertyValue("LoaderName", m_loader->name());
-  setProperty("LoaderVersion", m_loader->version());
+
+  /**
+   * Set the loader and version from correct loader IF one has been found,
+   * so that caller can reuse these variables for later calls.
+   * Though these are set in call to getFileLoader, they can be errantly
+   * changed by the init process .
+   */
+  if (m_loader) {
+    setPropertyValue("LoaderName", m_loader->name());
+    setProperty("LoaderVersion", m_loader->version());
+  }
 }
 
 void Load::loadSingleFile() {

--- a/Framework/DataHandling/test/LoadTest.h
+++ b/Framework/DataHandling/test/LoadTest.h
@@ -318,8 +318,8 @@ public:
     // verify that at the end, it is correctly set back according to the output
     loader.initialize();
     TS_ASSERT_THROWS_NOTHING(loader.setPropertyValue("Filename", "CNCS_7860_event.nxs"));
-    std::string correctLoader = loader.getPropertyValue("LoaderName");
-    std::string incorrectLoader = "NotALoader";
+    std::string const correctLoader = loader.getPropertyValue("LoaderName");
+    std::string const incorrectLoader = "NotALoader";
     TS_ASSERT_DIFFERS(correctLoader, incorrectLoader);
     TS_ASSERT_THROWS_NOTHING(loader.setPropertyValue("LoaderName", incorrectLoader));
     TS_ASSERT_THROWS_NOTHING(loader.execute());

--- a/Framework/DataHandling/test/LoadTest.h
+++ b/Framework/DataHandling/test/LoadTest.h
@@ -256,7 +256,7 @@ public:
     loader.initialize();
     loader.setPropertyValue("Filename", "084446+084447.nxs");
 
-    std::string outputWS = "LoadTest_out";
+    std::string outputWS = AnalysisDataService::Instance().uniqueName();
     loader.setPropertyValue("OutputWorkspace", outputWS);
     TS_ASSERT_THROWS_NOTHING(loader.execute());
 
@@ -279,7 +279,7 @@ public:
     loader.initialize();
     loader.setPropertyValue("Filename", "084446-084447");
 
-    std::string outputWS = "LoadTest_out";
+    std::string outputWS = AnalysisDataService::Instance().uniqueName();
     loader.setPropertyValue("OutputWorkspace", outputWS);
     TS_ASSERT_THROWS_NOTHING(loader.execute());
 
@@ -313,18 +313,30 @@ public:
   }
 
   void test_must_set_loadername() {
+    std::string const outputWS = AnalysisDataService::Instance().uniqueName();
+    std::string const incorrectLoader = "NotALoader";
+    int const incorrectVersion = -2;
+
     Load loader;
     // run Load with the LoaderName set to something
     // verify that at the end, it is correctly set back according to the output
     loader.initialize();
+    TS_ASSERT_THROWS_NOTHING(loader.setProperty("OutputWorkspace", outputWS))
     TS_ASSERT_THROWS_NOTHING(loader.setPropertyValue("Filename", "CNCS_7860_event.nxs"));
+    // the loader namer will be set: grab it an ensure it is not the bad cvalue
     std::string const correctLoader = loader.getPropertyValue("LoaderName");
-    std::string const incorrectLoader = "NotALoader";
+    int const correctVersion = loader.getProperty("LoaderVersion");
     TS_ASSERT_DIFFERS(correctLoader, incorrectLoader);
+    TS_ASSERT_DIFFERS(correctVersion, incorrectVersion);
+    // now SET the loader to a bad value, and execute
     TS_ASSERT_THROWS_NOTHING(loader.setPropertyValue("LoaderName", incorrectLoader));
+    TS_ASSERT_THROWS_NOTHING(loader.setProperty("LoaderVersion", incorrectVersion));
+    TS_ASSERT_EQUALS(loader.getPropertyValue("LoaderName"), incorrectLoader);
+    TS_ASSERT_EQUALS(loader.getProperty("LoaderVersion"), incorrectVersion);
     TS_ASSERT_THROWS_NOTHING(loader.execute());
     // make sure the loader name has been correctly set
     TS_ASSERT_EQUALS(loader.getPropertyValue("LoaderName"), correctLoader);
+    TS_ASSERT_EQUALS(loader.getProperty("LoaderVersion"), correctVersion);
   }
 };
 

--- a/Framework/DataHandling/test/LoadTest.h
+++ b/Framework/DataHandling/test/LoadTest.h
@@ -311,6 +311,21 @@ public:
     loader.setPropertyValue("Filename", "argus0026287.nxs");
     TS_ASSERT_EQUALS(loader.getPropertyValue("LoaderName"), "LoadMuonNexus");
   }
+
+  void test_must_set_loadername() {
+    Load loader;
+    // run Load with the LoaderName set to something
+    // verify that at the end, it is correctly set back according to the output
+    loader.initialize();
+    TS_ASSERT_THROWS_NOTHING(loader.setPropertyValue("Filename", "CNCS_7860_event.nxs"));
+    std::string correctLoader = loader.getPropertyValue("LoaderName");
+    std::string incorrectLoader = "NotALoader";
+    TS_ASSERT_DIFFERS(correctLoader, incorrectLoader);
+    TS_ASSERT_THROWS_NOTHING(loader.setPropertyValue("LoaderName", incorrectLoader));
+    TS_ASSERT_THROWS_NOTHING(loader.execute());
+    // make sure the loader name has been correctly set
+    TS_ASSERT_EQUALS(loader.getPropertyValue("LoaderName"), correctLoader);
+  }
 };
 
 //-------------------------------------------------------------------------------------------------

